prov/verbs/src/fi_verbs.c

```diff
@@ -557,110 +590,192 @@ static int fi_ibv_rai_to_fi(struct rdma_addrinfo *rai, struct fi_info *fi)
  	return 0;
 }
 
-static int fi_ibv_fill_info_attr(struct ibv_context *ctx, struct ibv_qp *qp,
-				 const struct fi_info *hints,
-				 struct fi_info *fi)
+static int fi_ibv_get_device_attrs(struct ibv_context *ctx,
+		const struct fi_info *hints, struct fi_info *info)
 {
-	struct ibv_qp_init_attr qp_init_attr;
+	struct ibv_pd *pd;
+	struct ibv_cq *cq;
+	struct ibv_qp *qp;
+	struct ibv_qp_init_attr init_attr;
 	struct ibv_device_attr device_attr;
 	struct ibv_port_attr port_attr;
-	struct ibv_qp_attr qp_attr;
-	union ibv_gid gid;
-	size_t name_len;
-	int ret;
-
-	*(fi->fabric_attr) = verbs_fabric_attr;
-	*(fi->domain_attr) = verbs_domain_attr;
-	*(fi->ep_attr)	   = verbs_ep_attr;
-	*(fi->tx_attr)	   = verbs_tx_attr;
-	*(fi->rx_attr)	   = verbs_rx_attr;
-	
-	if (!(fi->fabric_attr->prov_name = strdup(VERBS_PROV_NAME)))
-		return -FI_ENOMEM;
+	int ret = 0;
 
-	if (!ctx) {
-		if (!(fi->fabric_attr->name = strdup(VERBS_ANY_FABRIC)))
-			return -FI_ENOMEM;
+	pd = ibv_alloc_pd(ctx);
+	if (!pd)
+		return -errno;
 
-		return 0;
+	cq = ibv_create_cq(ctx, 1, NULL, NULL, 0);
+	if (!cq) {
+		ret = -errno;
+		goto err1;
 	}
 
-	if (qp) {
-		ret = ibv_query_qp(qp, &qp_attr, IBV_QP_CAP, &qp_init_attr);
-		if (ret)
-			return -ret;
-		fi->tx_attr->inject_size = qp_attr.cap.max_inline_data;
+	memset(&init_attr, 0, sizeof init_attr);
+	init_attr.send_cq = cq;
+	init_attr.recv_cq = cq;
+	init_attr.cap.max_send_wr = 1;
+	init_attr.cap.max_recv_wr = 1;
+	init_attr.cap.max_send_sge = 1;
+	init_attr.cap.max_recv_sge = 1;
```

Comment:
The qp is created here only to verify if hints->tx_attr->inject_size is accepted by the device. There isn't a way to get  max_inline_data directly. On the other hand, max_send_sge and max_recv_sge can be got from device_attr.max_sge. Hence there is no need to pass along hints->tx_attr->iov_limit here.